apps/storefront/src/pages/MyOrders/index.test.tsx

```diff
@@ -108,12 +109,16 @@ describe('when a personal customer', () => {
 
   describe('has placed orders', () => {
     it('displays a table with order information', async () => {
+      const getOrders = vi.fn();
```

Comment:
if you want the extra safety `vi.fn<unknown[], GetCustomerOrders>();`

it's stopped me from mocking the wrong returns now and then 🍹 

in ng-customers we ended up making a little helper for creating type safe mocks
we'd declare them once per test file, with their implementation being reset to self-destruct between tests
https://github.com/bigcommerce/ng-customers/blob/master/tests/utils/stub.ts

not sure if we're ready for that kind of stuff here, but it worked well with the `when`

Reply:
can do, I'll go and refactor on a follow up PR.